pyro/poutine/trace.py

```diff
@@ -20,103 +19,83 @@ def _warn_if_nan(name, value):
     # Note that -inf log_pdf is fine: it is merely a zero-probability event.
 
 
-class DiGraph(networkx.DiGraph):
-    node_dict_factory = collections.OrderedDict
-
-
 class Trace(object):
     """
-    Execution trace data structure
+    TODO docs
     """
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, graph_type=None):
         """
-        :param string graph_type: string specifying the kind of trace graph to construct
-
-        Constructor. Currently identical to networkx.``DiGraph(\*args, \**kwargs)``,
-        except for storing the graph_type attribute
+        TODO docs, args
         """
-        self._graph = DiGraph(*args, **kwargs)
-        graph_type = kwargs.pop("graph_type", "flat")
+        if graph_type is None:
+            graph_type = "flat"
         assert graph_type in ("flat", "dense"), \
             "{} not a valid graph type".format(graph_type)
         self.graph_type = graph_type
-        super(Trace, self).__init__(*args, **kwargs)
-
-    def __del__(self):
-        # Work around cyclic reference bugs in networkx.DiGraph
-        # See https://github.com/uber/pyro/issues/798
-        self._graph.__dict__.clear()
-
-    @property
-    def nodes(self):
-        return self._graph.nodes
-
-    @property
-    def edges(self):
-        return self._graph.edges
-
-    @property
-    def graph(self):
-        return self._graph.graph
-
-    @property
-    def remove_node(self):
-        return self._graph.remove_node
-
-    @property
-    def add_edge(self):
-        return self._graph.add_edge
+        # metadata
+        self.graph = collections.OrderedDict()
```

Comment:
what is `self.graph`?

Reply:
It's a collection of global metadata in a `networkx.DiGraph`.  I couldn't remember if it was used in Pyro, so I kept that part of the API, but will check now that things work.